perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -3565,18 +3581,32 @@ public void deleteAllAttributeAuthz(PerunSession sess, AttributeDefinition attri
 
 	@Override
 	public List<Attribute> getRequiredAttributes(PerunSession sess, Service service, Group group) throws InternalErrorException {
+		return getRequiredAttributes(sess, Collections.singletonList(service), group);
+	}
+
+	@Override
+	public List<Attribute> getRequiredAttributes(PerunSession sess, List<Service> services, Group group) throws InternalErrorException {
 		if(!CacheManager.isCacheDisabled() && !perun.getCacheManager().wasCacheUpdatedInTransaction()) {
-			List<Integer> attrIds = getRequiredAttributeIds(service);
+			List<Integer> attrIds = new ArrayList<>();
+			for (Service service : services) {
+				attrIds.addAll(getRequiredAttributeIds(service));
```

Comment:
Again we should have unique set of attribute ids.

Reply:
Done.